### PR TITLE
Backport changes to 0.18

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
List of backported PRs:
 * https://github.com/gtk-rs/gtk-rs-core/pull/1153
 * https://github.com/gtk-rs/gtk-rs-core/pull/1156
 * https://github.com/gtk-rs/gtk-rs-core/pull/1154
 * https://github.com/gtk-rs/gtk-rs-core/pull/1159
 * https://github.com/gtk-rs/gtk-rs-core/pull/1149
 * https://github.com/gtk-rs/gtk-rs-core/pull/1146
 * https://github.com/gtk-rs/gtk-rs-core/pull/1177
 * https://github.com/gtk-rs/gtk-rs-core/pull/1182
 * https://github.com/gtk-rs/gtk-rs-core/pull/1173

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
Cairo/sys also needs a release btw

**Comment by GuillaumeGomez** [comment]
Good catch! Updated it as well.

File: gio/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ authors = ["The gtk-rs Project Developers"]
 keywords = ["glib", "gio", "gtk-rs", "gnome"]
 readme = "README.md"
 documentation = "https://gtk-rs.org/gtk-rs-core/stable/latest/docs/gio/"
-version = "0.18.1"
+version = "0.18.2"
```
Comment (sdroege) [inline gio/Cargo.toml:9]: Please change version in all Cargo.toml, but only upload/publish the ones that have functional changes

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise. Change that one thing and go :) Thanks!

**Comment by GuillaumeGomez** [comment]
Done. So just need to publish (listing for myself):
 * cairo
 * cairo-sys
 * gio
 * glib
 * glib-macros

**Pull request merged**